pages/login/index.js

```diff
@@ -43,12 +63,35 @@ const Login = () => {
     Router.push('/signup')
   }
 
-  const discordAuth = () => {
-    dispatch(changeLoginType('discord'))
-    window.open(
-      `${process.env.NEXT_PUBLIC_MOON_SERVER_URL}/api/auth/discord`,
-      '_self'
-    )
+  const generateDiscordUrl = async () => {
+    try {
+      dispatch(authenticatePending())
+      await client.resetStore()
+      const res = await discordAuth()
+      if (res.data) {
+        openDiscordWindow(res.data.generateDiscordUrl)
+      }
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  const openDiscordWindow = (discordUrl) => {
+    const windowFeatures =
+      'height=800,width=800,resizable=yes,scrollbars=yes,status=yes'
+
+    const discordWindow = window.open(discordUrl, '_blank', windowFeatures)
+
+    const intervalId = setInterval(async () => {
+      if (discordWindow.closed) {
+        const session = await getSession()
+        if (session) {
```

Comment:
Add new line above

It's usually best practice to add 1 line of space in between statements to make code more readable

@micohalvarez